File: solutions/c/scrabble-score/1/scrabble_score.c
#include "scrabble_score.h"

#include <stdio.h>
#include <ctype.h>

int values[26] = {
    1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10
};

unsigned int score(const char *word) {
    unsigned int total = 0;
    for (int i = 0; word[i] != '\0'; i++) {
        char c = toupper(word[i]);
        if (c >= 'A' && c <= 'Z') {
            total += values[c - 'A'];
        }
    }
    return total;
}

File: solutions/lua/scrabble-score/1/scrabble-score.lua
local scores = { ['AEIOULNRST'] = 1, ['DG'] = 2, ['BCMP'] = 3, ['FHVWY'] = 4, ['K'] = 5, ['JX'] = 8, ['QZ'] = 10 }
return {
  score = function(word)
    local total = 0
    for ch in (word or ''):upper():gmatch(".") do
      for l, v in pairs(scores) do
        if l:match(ch) then
          total = total + v
          break
        end
      end
    end
    return total
  end
}
